shared/src/commonMain/kotlin/co/nimblehq/avishek/phong/kmmic/presentation/module/HomeViewModel.kt

```diff
@@ -117,6 +118,7 @@ class HomeViewModel(
     }
 
     private fun handleFetchMoreSurveySuccess(surveys: List<Survey>) {
+        this.surveys += surveys
         val surveyUiModels = surveys.map { SurveyUiModel(it) }
```

Comment:
We could do this instead and get rid of the `surveys` property.
```suggestion
        val surveyUiModels = _viewState.value.surveys + surveys.map { SurveyUiModel(it) }
```

Reply:
In this case, we need to store the list of `Survey` for passing data between the screen and the `SurveyUiModel` for display purposes. As Su.V's comment, passing the SurveyUiModel is not a good practice. And passing the survey's id requires an extra step to get the rest of the information for displaying on the Survey Detail screen. So I decided to pass an object of Survey.